packages/kas/src/nodes.ts

```diff
@@ -949,153 +981,145 @@ export class Mul extends Seq {
         }
     }
 
-    strip() {
-        var terms = _.map(this.terms, function (term) {
+    strip(): Expr {
+        var terms = _.map(this.terms, (term) => {
             return term instanceof Num ? term.abs() : term.strip();
         });
         return new Mul(terms).flatten();
     }
 
     // expand numerator and denominator separately
-    expand() {
-        var isAdd = function (term) {
+    expand(): Expr {
+        const isAdd = function (term: Expr): term is Add {
```

Comment:
`x instanceof Add` pops up a few times in the code. I wonder if it would be beneficial to make these predicates global and shared.

Reply:
I'm a little hesitant to extract predicates into their own top-level function if they're only used in a single place.  I know there's a cost to redefining functions like this, but I don't believe that KAS performance is current an issue.

Reply:
I'll defer to your judgement, but I'm a little confused by the "only used in a single place" part as `isAdd` is declared twice.

Reply:
You're right.  I'll hoist it.